ultet/src/main/scala/za/co/absa/ultet/parsers/PgFunctionFileParser.scala

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2023 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package za.co.absa.ultet.parsers
+
+import za.co.absa.ultet.dbitems.DBFunctionFromSource
+
+import java.net.URI
+import java.nio.file.{Files, Paths}
+import java.util.stream.Collectors
+
+case class PgFunctionFileParser() {
+
+  def parseFile(fileUri: URI): DBFunctionFromSource = {
+    val path = Paths.get(fileUri)
+
+    val lines = Files.lines(path)
+    val content = lines.collect(Collectors.joining("\n"))
+
+    parseString(content)
+  }
+
+  def parseString(str: String): DBFunctionFromSource = {
+    import PgFunctionFileParser._
+
+    val owner = ownerRx.findAllMatchIn(str).toList match {
+      case Nil => throw new IllegalStateException(s"Could not parse owner from $str")
+      case o :: Nil => o.group(1)
+      case _ => throw new IllegalStateException(s"Found more than 2 owners in $str")
+    }
```

Comment:
Would put each of these blocks into its own private function.

Reply:
Sure, thanks, done.